cmd/rollout-operator/main.go

```diff
@@ -152,6 +170,20 @@ func main() {
 		kubeConfig.UserAgent = rest.DefaultKubernetesUserAgent()
 	}
 
+	httpRT := http.DefaultTransport
+	if cfg.enableNamespaceValidation {
+		// HTTP client side cluster validation.
+		reporter := func(msg string, method string) {
+			level.Warn(logger).Log("msg", msg, "method", method, "cluster_validation_label", cfg.kubeNamespace)
+			metrics.InvalidClusterValidationLabels.WithLabelValues(method, "http", cfg.kubeNamespace).Inc()
+		}
+		httpRT = middleware.ClusterValidationRoundTripper(cfg.kubeNamespace, reporter, httpRT)
+
+		kubeConfig.Wrap(func(rt http.RoundTripper) http.RoundTripper {
+			return httpRT
+		})
```

Comment:
shouldn't we wrap the `rt` we get? why are we discarding it?

Reply:
Oops, good catch!

Reply:
Fixed.